src/components/Quiz/Quiz.tsx

```diff
@@ -0,0 +1,116 @@
+
+const images = {
+  'cat': kitty,
+  'dog': puppy
+}
+
+const allQuestions = {
+  'cat': catQuestions,
+  'dog': dogQuestions
+}
+
+interface QuizProps {
+  updateAnswers: (answer: string, queryNumber: number) => void
+  notifyReady: () => void
+}
+
+const Quiz = ({updateAnswers, notifyReady}: QuizProps) => {
+  const [selectedPet, setSelectedPet] = useState('cat')
+  const [question, setQuestion] = useState<Question>(dogOrCat)
+  const [sliderBar, setSliderBar] = useState<{visible: boolean, rating: string}>({visible: false, rating: ''})
+  const [ratingAnswer, setRatingAnswer] = useState('')
+  const [questionNumber, setQuestionNumber] = useState(0)
+
+
+  useEffect(() => {
+    const ratingAnswer = question.answers.find(answer => answer.value === sliderBar.rating)
+    if(ratingAnswer) setRatingAnswer(ratingAnswer.answer)
+  }, [sliderBar])
+  const updateSlider = (value:string) => {
+    setSliderBar({visible: true, rating: value})
+  }
+
+  useEffect(() => {
+    if(JSON.stringify(question) == JSON.stringify(dogOrCat)) {
+      setSliderBar({visible: false, rating: ''})
+    } else {
+      setSliderBar({visible: true, rating: '3'})
+    }
+  }, [question])
+
+  useEffect(() => {
+    if(questionNumber) {
+      setQuestion((allQuestions as Indexable)[selectedPet][questionNumber -1])
+    } else {
+      setQuestion(dogOrCat)
+    }
+  }, [questionNumber])
+  
+
+  const goToNextQuestion = () => {
+    if(questionNumber) {
+      updateAnswers(sliderBar.rating, questionNumber)
+    } else {
+      updateAnswers(selectedPet, questionNumber)
+    }
+    setQuestionNumber(prev => prev +1)
+  }
+
+  const goToPrevQuestion = () => setQuestionNumber(prev => prev -1)
+
+  const submitQuiz = () => {
+    updateAnswers(sliderBar.rating, questionNumber)
+    notifyReady()
+  }
+  
+  const animalEls = question.answers.map(answer => {
+    return (
+    <section key={answer.value} className={selectedPet === answer.value ? 'selected pet-choice' : 'pet-choice'} onClick={() => setSelectedPet(answer.value)}>
+      <img src={(images as Indexable)[answer.value]} alt={`${answer.value}`}/>
+      <p>{answer.answer}</p>
+    </section>
+    )
+  })
+
+  return (
+    <section className='quiz'>
+      {questionNumber > 0 && <button onClick={goToPrevQuestion} className='back-btn'><img src={back} alt='back button' /></button>}
+      <header className='quiz-header'>
+        <h1 className='question header-child'>{question.question}</h1>
+        {sliderBar.visible && <p className='instructions header-child'>slide the rating bar to change your answer</p>}
+      </header>
+      {sliderBar.visible && 
+      <>
+        <div className='slider-bar-container'>
+          <input type='range' value={sliderBar.rating} onChange={(e) => setSliderBar({visible: true, rating: e.target.value})} min='1' max='5'/>
+          <div className='range-markers'>
+            <span className='marker' id='marker1' onClick={() => updateSlider('1')}><p>1</p></span>
```

Comment:
This was really cool to see how you did the slider bar!! I was definitely curious about it